Sources/SourceKitLSP/Swift/MacroExpansionEdit.swift

```diff
@@ -0,0 +1,57 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import LanguageServerProtocol
+import SourceKitD
+
+/// Represents a macro expansion as an edit. Notionally, a subclass of `TextEdit`
+public struct MacroExpansionEdit: ResponseType, Hashable, Sendable {
```

Comment:
Does this type have to be `public`? Also, I don’t think it has to conform to `ResponseType` because we never return it as a response to an LSP request – we always encode it to `LSPAny` first.

Reply:
Yeah right

Reply:
Wait, I realised that the tests are using the `MacroExpansionEdit`. So, it has to be `public`? or should we not mark this `public` and try some workarounds in the tests?

Reply:
Ah, I missed that. Let’s mark it as `@_spi(Testing) public` then and `@_spi(Testing) import SourceKitLSP` in the test.